azurerm/internal/services/synapse/tests/synapse_role_assignment_resource_test.go

```diff
@@ -0,0 +1,188 @@
+	workspaceClient := acceptance.AzureProvider.Meta().(*clients.Client).Synapse.WorkspaceClient
+	environment := acceptance.AzureProvider.Meta().(*clients.Client).Account.Environment
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_synapse_role_assignment" {
+			continue
+		}
+		id, err := parse.SynapseRoleAssignmentID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+		if resp, err := workspaceClient.Get(ctx, id.Workspace.ResourceGroup, id.Workspace.Name); err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("bad: Get on Synapse.WorkspaceClient %q, %+v", id.Workspace.String(), err)
+			}
+			return nil
+		}
+
+		client := synapseClient.AccessControlClient(id.Workspace.Name, environment.SynapseEndpointSuffix)
+		resp, err := client.GetRoleAssignmentByID(ctx, id.Id)
+		if err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("bad: Get on Synapse.AccessControlClient: %+v", err)
+			}
+			return nil
+		}
+		return fmt.Errorf("expected no Synapse Role Assignment but found %+v", resp)
+	}
+	return nil
+}
+
+func testAccAzureRMSynapseRoleAssignment_basic(data acceptance.TestData) string {
+	template := testAccAzureRMSynapseRoleAssignment_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_synapse_role_assignment" "test" {
+  synapse_workspace_id = azurerm_synapse_workspace.test.id
+  role_name            = "Sql Admin"
+  principal_id         = data.azurerm_client_config.current.object_id
+}
+`, template)
+}
+
+func testAccAzureRMSynapseRoleAssignment_requiresImport(data acceptance.TestData) string {
+	config := testAccAzureRMSynapseRoleAssignment_basic(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_synapse_role_assignment" "import" {
+  synapse_workspace_id = azurerm_synapse_role_assignment.test.synapse_workspace_id
+  role_name            = azurerm_synapse_role_assignment.test.role_name
+  principal_id         = azurerm_synapse_role_assignment.test.principal_id
+}
+`, config)
+}
+
+func testAccAzureRMSynapseRoleAssignment_template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctest-Synapse-%d"
+  location = "%s"
+}
+
+resource "azurerm_storage_account" "test" {
+  name                     = "acctestacc%s"
+  resource_group_name      = azurerm_resource_group.test.name
+  location                 = azurerm_resource_group.test.location
+  account_kind             = "BlobStorage"
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+}
+
+resource "azurerm_storage_data_lake_gen2_filesystem" "test" {
+  name               = "acctest-%d"
+  storage_account_id = azurerm_storage_account.test.id
+}
+
+resource "azurerm_synapse_workspace" "test" {
+  name                                 = "acctestsw%d"
```

Comment:
could we use capitals hereif allowed to ensure the API handles them correctly?

Reply:
synapse workspace name could only contain lowercase letters or numbers. We already have this validation rule